packages/forms/src/pattern.ts

```diff
@@ -133,6 +133,27 @@ export const validatePattern = (
   return r.success(parseResult.data);
 };
 
+const aggregateValuesByPrefix = (
+  values: Record<string, string>
+): Record<string, any> => {
+  const aggregatedValues: Record<string, any> = {};
+
+  for (const [key, value] of Object.entries(values)) {
+    const [prefix, suffix] = key.split('.');
```

Comment:
Glad you figured out what was going on here!

I think we may want to do something like this, to deal with more deeply nested objects, such as `abc.def.0.xyz`:

```suggestion
    const [prefix, ...suffix] = key.split('.');
```

Then we'd need to handle the suffix as an array, perhaps by making this a recursive function...

There are also several libraries that do this, which may be easier. This one is already in our lock file:

https://www.npmjs.com/package/set-value

And this one is also heavily used:

https://www.npmjs.com/package/dot-prop

Reply:
Good point! Updating it to set-value.

Reply:
Updated!